scripts/gen_lsp.lua

```diff
@@ -1,11 +1,15 @@
---[[
-Generates lua-ls annotations for lsp
+-- Generates lua-ls annotations for lsp.
+
+local USAGE = [[
+Generates lua-ls annotations for lsp.
+
 USAGE:
-nvim -l scripts/gen_lsp.lua gen  # this will overwrite runtime/lua/vim/lsp/_meta/protocol.lua
-nvim -l scripts/gen_lsp.lua gen --version 3.18 --build/new_lsp_types.lua
+nvim -l scripts/gen_lsp.lua gen  # by default, this will overwrite runtime/lua/vim/lsp/_meta/protocol.lua
 nvim -l scripts/gen_lsp.lua gen --version 3.18 --out runtime/lua/vim/lsp/_meta/protocol.lua
 nvim -l scripts/gen_lsp.lua gen --version 3.18 --methods
---]]
+]]
+
+local DEFAULT_LSP_VERSION = '3.18'
```

Comment:
Maybe we need to set 3.17 (Current) at this moment -- https://microsoft.github.io/language-server-protocol/specifications/lsp/3.18/specification/ is "Upcoming" (still changing).

One issue is that [the current JSON data in the 3.17](https://github.com/microsoft/language-server-protocol/blob/gh-pages/_specifications/lsp/3.17/metaModel/metaModel.json?rgh-link-date=2023-12-11T02%3A19%3A24Z) specs happened to include pending ones for 3.18  (see https://github.com/microsoft/language-server-protocol/pull/1816)

Reply:
🤷